profiler/profiler.go

```diff
@@ -92,12 +96,18 @@ func (p *profiler) run() {
 	if _, ok := p.cfg.types[BlockProfile]; ok {
 		runtime.SetBlockProfileRate(p.cfg.blockRate)
 	}
+	p.wg.Add(1)
 	go func() {
 		tick := time.NewTicker(p.cfg.period)
 		defer tick.Stop()
 		p.collect(tick.C)
+		p.wg.Done()
```

Comment:
Can we put these on the first line in a `defer`? They've always been coupled together with their counterparts.